portfolio/src/main/webapp/quiz.js

```diff
@@ -0,0 +1,58 @@
+
+
+function makeQuiz(){
+    questions.forEach((currentQuestion,questionNumber)=>{
+        output.push(
+            `<input type="checkbox" id="question${questionNumber}" name="question${questionNumber}" 
+            value="yes">
+            <label for="question${questionNumber}">${currentQuestion}</label></br>`);
+    });
+    quizContainer.innerHTML=output.join('');
+}
+function showResults(){
+    let count = 0;
+    questions.forEach( (currentQuestion, questionNumber) => {
+        let currentValue = document.getElementById(`question${questionNumber}`);
+        if(currentValue.checked === true){
+            count++;
+        }
+    });
+    let resultsCode = `<h2>Your Results:</h2><p>Your score was ${count}</p>`;
+
+    if(count<= 1){
+        resultsCode += `<p>You are definitely not a cat. </p>`;
+    }else if (count <=7){
+        resultsCode += `<p>You are not a cat, but you do have some catlike tendencies.</p>`;
+    }else if (count <=10){
+        resultsCode += `<p>You might be a cat.</p>`;
+    }else{
+        resultsCode += `<p>You are a cat.</p>`;
+    }
+    resultsContainer.innerHTML = resultsCode;
```

Comment:
It's safer to embed these tags directly into the HTML, and then use [node.textContent](https://developer.mozilla.org/en-US/docs/Web/API/Node/textContent) to replace the text of those nodes programmatically. This is because if user-generated content slips into the HTML, then you could be vulnerable to attacks like XSS: https://wiki.developer.mozilla.org/en-US/docs/Glossary/Cross-site_scripting

For the `<h2>` tag, you can set the display styles to show/hide it: https://developer.mozilla.org/en-US/docs/Web/CSS/visibility

Another alternative is to create DOM nodes in JavaScript, and then add them as children to resultsContainer. [See the example.](https://developer.mozilla.org/en-US/docs/Web/API/Node/appendChild)

Reply:
Acknowledged